### DISCUSSION TITLE
Relay Server Compliance CRUD

### DISCUSSION THREAD

**Discussion by DaniloOliveira28** [sibelius/relay-workshop-old#5] [General]
What are the best practices/rules that we need to implement in the server in order to attend a relay client in a CRUD operation?

Are these definitions already defined in some place?

**Comment by sibelius** [comment]
You should follow Updaters configs idea https://relay.dev/docs/v10.1.3/mutations#updater-configs

For a mutation that removes a node you need to return the deletedID

```graphql
  mutation DestroyShipMutation($target: ID!) {
    destroyShip(target: $target) {
      destroyedShipId      
    }
  }
```

You could also use the new declarative directives to make this easier and avoid writing an updater

https://relay.dev/docs/guided-tour/list-data/updating-connections/#using-declarative-deletion-directive

```graphql
mutation DeleteCommentsMutation(
      # Define a GraphQL variable for the connections array
      $connections: [ID!]!
      $input: CommentsDeleteInput
    ) {
      commentsDelete(input: $input) {
        deletedCommentIds @deleteEdge(connections: $connections)
      }
    }
```

**Comment by arantespp** [comment]
1. You need to implement [node interface](https://relay.dev/docs/guides/graphql-server-specification/#object-identification) in every type of your schema. In other words, every node must have an unique ID.
2. If you want to use pagination features, you must implement [connections schema](https://dev.to/ttoss/lett-s-go-build-pagination-with-relay-and-react-an-intermediary-tutorial-f89).
